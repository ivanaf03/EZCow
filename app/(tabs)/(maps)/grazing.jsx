import React from "react";

import { View, SafeAreaView, Text, FlatList } from "react-native";
import { router } from "expo-router";

import CustomButton from "../../../components/basic/custom-button";
import TabTitle from "../../../components/tabs/tab-title";
import { useUser } from "../../../store/user-provider";
import {
  getExitedGrazingByuserId,
  getGrazingByuserId,
  expireGrazing,
} from "../../../model/grazing";
import GrazingCard from "./grazing-card";

const Grazing = () => {
  const [grazing, setGrazing] = React.useState([]);
  const [exitedGrazing, setExitedGrazing] = React.useState([]);

  const { user } = useUser();

  React.useEffect(() => {
    loadGrazing();
    loadExitedGrazing();
  }, []);

  const loadGrazing = async () => {
    const res = await getGrazingByuserId(user.id);
    setGrazing(res);
  };

  const loadExitedGrazing = async () => {
    const res = await getExitedGrazingByuserId(user.id);
    setExitedGrazing(res);
  };

  const deleteGrazing = async (grazing) => {
    await expireGrazing(grazing.grazingId);
    await loadGrazing();
    await loadExitedGrazing();
  };

  return (
    <SafeAreaView className="flex-1 bg-c_dark_gray">
      <TabTitle text="Ver pastoreo" />
      <View className="mx-4 mt-4 space-y-2">
        <View>
          <CustomButton
            text="Añadir al pastoreo"
            onPress={() => router.replace("grazing-form")}
            buttonTestID="grazing-form-button"
          />
        </View>
        <View className="flex-row items-center">
          <View className="w-[50%] pr-2">
            <CustomButton
              text="Ver ubicaciones"
              onPress={() => router.replace("fields")}
              buttonTestID={"fields-button"}
            />
          </View>
          <View className="w-[50%] pl-2">
            <CustomButton
              text="Ver parcelas"
              onPress={() => router.replace("cadastral-fields")}
              buttonTestID={"cadastral-fields-button"}
            />
          </View>
        </View>
      </View>
      <View className="flex-1">
        <View className="h-[50%]">
          <Text className="py-2 mx-2 font-Nunito_SemiBold text-c_white text-2xl border-b-2 border-c_white">
            Pastoreo actual
          </Text>
          <View>
            <FlatList
              data={grazing}
              keyExtractor={(item) => item.grazingId}
              renderItem={({ item }) => (
                <GrazingCard
                  grazing={item}
                  onDelete={deleteGrazing}
                  isExited={false}
                />
              )}
              showsVerticalScrollIndicator={true}
            />
          </View>
        </View>
        <View className="mt-12 pb-12 h-[40%]">
          <Text className="py-2 mx-2 font-Nunito_SemiBold text-c_white text-2xl border-b-2 border-c_white">
            Histórico
          </Text>
          <View className>
            <FlatList
              data={exitedGrazing}
              keyExtractor={(item) => item.grazingId}
              renderItem={({ item }) => (
                <GrazingCard
                  grazing={item}
                  onDelete={deleteGrazing}
                  isExited={true}
                />
              )}
              showsVerticalScrollIndicator={true}
            />
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};

export default Grazing;
