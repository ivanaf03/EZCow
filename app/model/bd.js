import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
