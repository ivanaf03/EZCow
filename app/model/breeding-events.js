import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};