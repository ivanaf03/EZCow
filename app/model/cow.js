import { getDatabase } from './bd';

export const insertCow = async (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) => {

    db = await getDatabase();

    const res = await db.runAsync(
        `INSERT INTO Cow (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`,
        [code, name, entryDate, gender, breed, phase, user_fk, mother_fk]
    );

    return res;
};

export const getCowIdByCode = async (code) => {

    db = await getDatabase();

    const res = await db.getFirstAsync(
        `SELECT id FROM Cow WHERE code = ?`,
        [code]
    );

    return res.id;
};

export const getAvailableCowNameById = async (id) => {

    db = await getDatabase();

    const res = await db.getFirstAsync(
        `SELECT name FROM Cow WHERE id = ? AND exitDate is NULL`,
        [id]
    );

    return res.name;
};

export const getAvailableCowCodeByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT code FROM Cow WHERE user_fk = ? AND exitDate is NULL AND gender='Femenino' AND phase='Adulto'`,
        [userId]
    );

    return res.map((cow) => cow.code);
};

export const getAllCowsAvailableByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
        [userId]
    );

    return res;
};

export const getAllCowsAvailableByUserIdAndPhaseAndGender = async (userId, phase, gender) => {
    
    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = ? AND gender = ?`,
        [userId, phase, gender]
    );

    return res;
};

export const getCalvesAvailableByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = 'Ternero'`,
        [userId]
    );

    return res;
};

export const setExitDateByCowId = async (cowId) => {

    db = await getDatabase();


    const res = await db.runAsync(
        `UPDATE Cow SET exitDate = CURRENT_DATE WHERE id = ?`,
        [cowId]
    );

    return res;
};

export const getAllCowsExitedByUserIdAndPhaseAndGender = async (userId, phase, gender) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = ? AND gender = ?`,
        [userId, phase, gender]
    );

    return res;
};

export const getAllCowsExitedByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL`,
        [userId]
    );

    return res;
};

export const getAllCalvesExitedByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = 'Ternero'`,
        [userId]
    );

    return res;
};