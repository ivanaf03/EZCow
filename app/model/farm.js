import { getDatabase } from "./bd";

export const insertFarmUbication = async (userId, latitude, longitude) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE User SET latitude = ?, longitude = ? WHERE id = ?`,
      [latitude, longitude, userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFarmUbicationByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT latitude, longitude FROM User WHERE id = ?`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
