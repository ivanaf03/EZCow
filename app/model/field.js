import { getDatabase } from "./bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
