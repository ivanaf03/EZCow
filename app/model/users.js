import CryptoJS from "crypto-js";
import { getDatabase } from "./bd";

const hashPassword = (password) => {
  return CryptoJS.SHA256(password).toString();
};

export const insertUser = async (name, email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `INSERT INTO User (name, email, password) VALUES (?, ?, ?)`,
      [name, email, hashedPassword]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertUserGoogle = async (name, email, google_id) => {
  const db = await getDatabase();
  try {
    const existingUser = await db.getFirstAsync(
      `SELECT * FROM User WHERE email = ?`,
      [email]
    );
    if (existingUser) {
      return existingUser;
    }
    const res = await db.runAsync(
      `INSERT INTO User (name, email, google_id) VALUES (?, ?, ?)`,
      [name, email, google_id]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getUserByEmail = async (email) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT * FROM User WHERE email = ?`, [
      email,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const changePassword = async (email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `UPDATE User SET password = ? WHERE email = ?`,
      [hashedPassword, email]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
