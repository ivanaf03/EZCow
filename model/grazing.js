import { getDatabase } from "./bd";

export const insertGroup = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`INSERT INTO CowGroup (name) VALUES (?)`, [
      name,
    ]);
    return res.lastID;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllGroups = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT DISTINCT CowGroup.id AS groupId,CowGroup.name AS groupName
       FROM CowGroup JOIN Cow ON CowGroup.id = Cow.group_fk WHERE Cow.user_fk = ? AND CowGroup.exitDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const setExitDateByGroupId = async (groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE CowGroup SET exitDate = CURRENT_DATE WHERE id = ?`,
      [groupId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertCowInGroup = async (cowId, groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`UPDATE Cow SET group_fk = ? WHERE id = ?`, [
      groupId,
      cowId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getCowsInGroup = async (groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Cow.id AS cowId,Cow.name AS cowName,Cow.code AS cowCode,Cow.group_fk AS groupId
       FROM Cow WHERE group_fk = ?`,
      [groupId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertGrazing = async (fieldId, groupId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`INSERT INTO Grazing (field_fk, group_fk, initialDate) VALUES (?, ?, CURRENT_DATE)`, [
      fieldId,
      groupId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFieldIdByName = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT id FROM Field WHERE name = ?`,
      [name]
    );
    return res.id;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFieldNames = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT name FROM Field WHERE user_fk = ?`,
      [userId]
    );
    return res.map((field) => field.name);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGroupIdByName = async (name) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT id FROM CowGroup WHERE name = ?`,
      [name]
    );
    return res.id;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGroupNames = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT DISTINCT CowGroup.name AS groupName FROM CowGroup JOIN Cow ON CowGroup.id = Cow.group_fk WHERE Cow.user_fk = ? AND CowGroup.exitDate IS NULL`,
      [userId]
    );
    return res.map((group) => group.groupName);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getGrazingByuserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Grazing.id AS grazingId,Grazing.initialDate AS initialDate,Grazing.endDate AS endDate,Field.name AS fieldName,CowGroup.name AS groupName FROM Grazing JOIN Field ON Grazing.field_fk = Field.id JOIN CowGroup ON Grazing.group_fk = CowGroup.id WHERE Field.user_fk = ? AND Grazing.endDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getExitedGrazingByuserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT Grazing.id AS grazingId,Grazing.initialDate AS initialDate,Grazing.endDate AS endDate,Field.name AS fieldName,CowGroup.name AS groupName FROM Grazing JOIN Field ON Grazing.field_fk = Field.id JOIN CowGroup ON Grazing.group_fk = CowGroup.id WHERE Field.user_fk = ? AND Grazing.endDate IS NOT NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const expireGrazing = async (grazingId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(`UPDATE Grazing SET endDate = CURRENT_DATE WHERE id = ?`, [
      grazingId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};