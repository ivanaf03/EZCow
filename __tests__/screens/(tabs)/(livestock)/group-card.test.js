import React from "react";

import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { router } from "expo-router";
import { Alert } from "react-native";

import GroupCard from "../../../../app/(tabs)/(livestock)/group-card";
import { insertGroup } from "../../../../model/grazing";

jest.mock("expo-router", () => ({
  router: {
    replace: jest.fn(),
  },
}));

jest.mock("../../../../store/user-provider", () => ({
  useUser: jest.fn(() => ({
    user: { id: "1", name: "TestUser", email: "test@test.com" },
  })),
}));

jest.mock("../../../../model/grazing", () => ({
  insertGroup: jest.fn(),
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

jest.spyOn(Alert, "alert");

describe("GroupCard", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = render(<GroupCard group={{groupId: 1, groupName: "TestGroup"}} onDelete={jest.fn()} />);
    await waitFor(() => expect(tree).toMatchSnapshot());
  });

  it("should call onDelete when pressing delete button", async () => {
    const onDelete = jest.fn();
    const { getByTestId } = render(<GroupCard group={{groupId: 1, groupName: "TestGroup"}} onDelete={onDelete} />);
    await waitFor(() => expect(getByTestId("delete-group-button")).toBeTruthy());
    const handleDeleteGroupButton = getByTestId("delete-group-button");
    fireEvent.press(handleDeleteGroupButton);
    expect(onDelete).toHaveBeenCalledWith({groupId: 1, groupName: "TestGroup"});
  });
});