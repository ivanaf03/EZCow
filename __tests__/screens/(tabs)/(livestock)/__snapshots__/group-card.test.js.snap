// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupCard should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "marginBottom": 16,
        "marginTop": 16,
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
        {
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  >
    TestGroup
  </Text>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "backgroundColor": "#272727",
          },
          {
            "borderBottomWidth": 2,
            "borderLeftWidth": 2,
            "borderRightWidth": 2,
            "borderTopWidth": 2,
          },
          {
            "borderBottomColor": "#6bc0eb",
            "borderLeftColor": "#6bc0eb",
            "borderRightColor": "#6bc0eb",
            "borderTopColor": "#6bc0eb",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
        ]
      }
      testID="delete-group-button"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Nunito-SemiBold",
            },
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            [
              {
                "marginLeft": 16,
              },
            ],
          ]
        }
      >
        Borrar
      </Text>
    </View>
  </View>
</View>
`;
